### ISSUE TITLE
ExecuteAsync method not returning response

### ISSUE THREAD

**Issue by karen-hikesupport** [restsharp/RestSharp#1906]
Version : 108.0.1.0

Currently we upgraded the restShap to project and change the method Execute to ExecuteAsync for Async method. but When i call executeasync method it is not returning reponse and some where the same executeAsync is work. 

Not getting an idea, where the actual issue, it is working fine when replace Execute method.

**Comment by RalfWenzel** [comment]
I can confirm this error or at least a similiar situation. Background: The code below is called at a high frequency. The code expects a HTTP status code and content to be returned. This is the case all the time, but from time to time content is null. I fist assumed that this may be a problem on the server side but then realized that ``restResponse.StatusCode`` is = 0 in this cases what should never happen.
```
public async Task<HecResponseMessage> PostServicesCollector(List<string> jsonPayload, string defaultHost = null, string defaultIndex=null, 
            string defaultSource = null, string defaultSourcetype = null)
        {
            if (jsonPayload == null || jsonPayload.Count < 1) throw new InvalidOperationException("Payload null or empty.");
            var request = new RestRequest("services/collector", Method.Post);
            if (defaultHost != null) request.AddParameter("host", defaultHost);
            if (defaultIndex != null) request.AddParameter("index", defaultIndex);
            if (defaultSource != null) request.AddParameter("source", defaultSource);
            if (defaultSourcetype != null) request.AddParameter("sourcetype", defaultSourcetype);
            StringBuilder sb = new StringBuilder();
            foreach (string record in jsonPayload)
            {
                sb.AppendLine(record);
            }
            sb.AppendLine();
            string body = sb.ToString();
            request.AddParameter("application/text", body, ParameterType.RequestBody);

            //RestResponse restResponse = await restClient.PostAsync(request);
            RestResponse restResponse = await restClient.ExecutePostAsync(request);
            if(string.IsNullOrEmpty(restResponse.Content))
            {
                if (_logger != null) _logger.LogError($"PostServicesCollector({jsonPayload.Count} records): HTTP Status {restResponse.StatusCode}. No content.");
                HecResponseMessage tmpResponse = new HecResponseMessage()
                {
                    Code = 0,
                    Text = "Generated",
                    InvalidEventNumber = -1
                };
                tmpResponse.Validate(HttpStatusCode.OK);
                return tmpResponse;
            }
            HecResponseMessage hecResponse = JsonConvert.DeserializeObject<HecResponseMessage>(restResponse.Content);
            hecResponse.Validate(restResponse.StatusCode);
            return hecResponse;
        }
````

**Comment by karen-hikesupport** [comment]
Hi @RalfWenzel 

Here the same situation with below line 

RestResponse restResponse = await restClient.ExecutePostAsync(request);

it is not returning the any response with above line but when i replace with exevute method withou async and await. it will work.

**Comment by RalfWenzel** [comment]
hi @karen-hikesupport ,

thx for response. I thought about testing it also but it would slow down my app because it is designed for async operations.
Currently i hope for a response from the maintainers but I'm looking for an alternative to replace restsharp also because i cannot go productive with my app because of this.

**Comment by RalfWenzel** [comment]
@karen-hikesupport let me add that i testet also with 108.0.0 -> same results

**Comment by josercsouza** [comment]
I am using 107.3 with Net 6 and I have the same problem when I used on production enviroment in AWS Windows +  IIS, but when I run in the my VS 2022 debug using IIS, working fine. I tried to find a solution but nothing work.

var response = clientRest.ExecuteAsync(_request).GetAwaiter().GetResult();

Using version before 107.3 was working fine, but as see that all working fine in dev, I hope that in PROD will be ok... :(
I disabled the firewall but the error still happening.
